Sources/MuxUploadSDK/Public API/MuxUpload+Transport.swift

```diff
@@ -0,0 +1,24 @@
+//
+//  MuxUpload+Transport.swift
+//
+
+import Foundation
+
+extension MuxUpload {
+
+    /// Transport-related settings
+    public struct TransportSettings {
```

Comment:
As I am plumbing this through the SDK, had a thought of why not go a step further and consolidate `TransportSettings` and `StandardizationSettings` into one structure:

```swift
struct UploadSettings {
    var transport: Transport

    struct Transport {
        // same fields as extant TransportSettings
    }

    enum Standardization {
       // same as extant StandardizationSettings
    }

    // adding an extra one
    enum Reporting {
       // opt out for events goes here?
    }
}
```

Any opinions?

Reply:
I think a big struct would be better. Consolidating the settings should improve config discoverability, which is good for devex

Reply:
I like this grouping. Keeping them together makes sense to me 👍

Reply:
[6876978](https://github.com/muxinc/swift-upload-sdk/pull/17/commits/687697880f9b6669544abf763c84208bee436663)